### PR TITLE
[KOGITO-8301] Documentation for Job Service - Leader instance election

### PR DESCRIPTION
**Pull request by tiagodolphine**
**JIRA:**

**Description:**

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] The nav.adoc file has a link to this guide in the proper category
- [ ] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc [hunk 1]
```diff
@@ -213,11 +214,37 @@ The example shows how to run PostgreSQL as a Kubernetes deployment, but you can
 ==== Ephemeral
 Alternatively, there is an in-memory database support that does not require any external database configuration, it can be used for testing and development purposes, but it is not recommended for production, since all jobs are lost upon a service restart or failure.
 
-====
 [NOTE]
 ====
 The job service link:{jobs_service_image_url}[image] is equipped with the PostgreSQL and Ephemeral (InMemory) options that can be switched using the `JOBS_SERVICE_PERSISTENCE`. If not set, it defaults to the `ephemeral` option. For more information about the job service, the container image can be found link:{jobs_service_image_usage_url}(here).
 ====
+
+=== Job Service Leader election feature
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:222]: We can remove the `feature` word from the title.
```suggestion
=== Job service leader election
```

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc [hunk 2]
```diff
@@ -213,11 +214,37 @@ The example shows how to run PostgreSQL as a Kubernetes deployment, but you can
...
+
+Currently, job service works in a single instance manner where there should be just one active instance of the service.
+
+To avoid issues when deploying the service into the cloud where it is common to eventually have more than one instance deployed, job service supports a leader instance election process, where an instance once became the leader, activates the external communication to receive and schedule jobs internally.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:226]: +1 @wmedvede

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc [hunk 3]
```diff
@@ -213,11 +214,37 @@ The example shows how to run PostgreSQL as a Kubernetes deployment, but you can
...
+
+.Job Service leader election
+image::core/job-service-leader.png[]
+
+[NOTE]
+====
+This leader election mechanism uses the underlying persistence backend, which currently is only supported in the PostgreSQL implementation.
+====
+
+There is no need for any configuration to support this feature, the only requirement is to have the supported database with the data schema
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:242]:
```suggestion
There is no need for any configuration to support this feature, the only requirement is to have the supported database with the data schema up-to-date as described in <<job-service-postgresql>>.
```

**Review by kaldesai — APPROVED** [review]
@tiagodolphine Very minor suggestions! Otherwise, your content looks good to me.

**Review by ricardozanini — COMMENTED** [review]
Many thanks, looks neat!
Although I understand it's not in the scope of this PR, can you please do the following modifications to this guide?

- Move the `Timeouts Showcase` content to a new guide under `Use Cases`
- Review the chapter hierarchy (the `==` things) to look correctly. See the attached screenshot.

Screenshot 2023-01-30 at 12 01 09


Thank you!

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:226]: +1

File: serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc [hunk 4]
```diff
@@ -213,11 +214,37 @@ The example shows how to run PostgreSQL as a Kubernetes deployment, but you can
 ==== Ephemeral
 Alternatively, there is an in-memory database support that does not require any external database configuration, it can be used for testing and development purposes, but it is not recommended for production, since all jobs are lost upon a service restart or failure.
 
-====
 [NOTE]
```
[the remark this answers is not in the document]
Comment (tiagodolphine) [reply serverlessworkflow/modules/ROOT/pages/core/timeouts-support.adoc:217]: +1

**Review by cristianonicolai — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**